scripts/exregional_run_fcst.sh

```diff
@@ -502,15 +502,263 @@ else
   fi
 fi
 
+#
+#-----------------------------------------------------------------------
+#
+# Set the stochatics physics related namelist on the fly for the current
+# forecast cycle
+#
+#-----------------------------------------------------------------------
+#
+hh=${cdate:8:2}
+for cyc_start in "${CYCL_HRS_SPP[@]}"; do
+  if [ ${hh} -eq ${cyc_start} ]; then
+    DO_SPP=TRUE
+  fi
+done
+for cyc_start in "${CYCL_HRS_SPPT[@]}"; do
+  if [ ${hh} -eq ${cyc_start} ]; then
+    DO_SPPT=TRUE
+  fi
+done
+for cyc_start in "${CYCL_HRS_SHUM[@]}"; do
+  if [ ${hh} -eq ${cyc_start} ]; then
+    DO_SHUM=TRUE
+  fi
+done
+for cyc_start in "${CYCL_HRS_SKEB[@]}"; do
+  if [ ${hh} -eq ${cyc_start} ]; then
+    DO_SKEB=TRUE
+  fi
+done
+for cyc_start in "${CYCL_HRS_LSM_SPP[@]}"; do
+  if [ ${hh} -eq ${cyc_start} ]; then
+    DO_LSM_SPP=TRUE
+  fi
+done
+#
```

Comment:
@chunhuazhou, do you plan on running with each of these schemes, or is this just a placeholder for now?  Also, there's a typo in "stochastic" above.

Reply:
I'm also a bit confused, because the DO_* stochastic options should go in the config.sh (now config.yaml) script, not in an ex-script.  I think it would make more sense to have config-level options like DO_SPP_ENSINIT or DO_SPP_SPINUP = TRUE, but maybe that's too complicated to implement at this time.

Reply:
> 

This is more of a placeholder for now. Thanks for pointing out the typo - corrected.

Reply:
> I'm also a bit confused, because the DO_* stochastic options should go in the config.sh (now config.yaml) script, not in an ex-script. I think it would make more sense to have config-level options like DO_SPP_ENSINIT or DO_SPP_SPINUP = TRUE, but maybe that's too complicated to implement at this time.

Yes I agree it is a bit confused. The reason I made these changes is for when we don't want to run the same setting for all the cycles. For example, in our RRFS ensemble cycling, we don't run SPP for the DA cycling hours (18-23Z) but we do want to turn SPP on for 00Z free forecast. Therefore we want to have those stochastic physics setting specific for each forecast cycle.